### PR TITLE
Add VerifyTests

### PR DESCRIPTION
**Pull request by rstm-sf**
Testsing with https://github.com/VerifyTests/Verify

### CODE REVIEW

**Comment by PrzemyslawKlys** [comment]
Hi, 

I saw you playing with this, and this is an excellent idea. You will need to add some "how to" so it can be easily regenerated if needed so I can do it myself from time to time ;)

As for the name of the pipeline I am not sure where it's coming from:

image

Do you know where?

File: OfficeIMO.Tests/OfficeIMO.Tests.csproj [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFrameworks Condition=" '$([MSBuild]::IsOsPlatform(`Windows`))' ">net472;net48;netcoreapp3.1;net6.0;net7.0</TargetFrameworks>
-        <TargetFrameworks Condition=" '$([MSBuild]::IsOsPlatform(`OSX`))' Or '$([MSBuild]::IsOsPlatform(`Linux`))' ">net6.0;net7.0</TargetFrameworks>
+        <TargetFrameworks>net6.0;net7.0</TargetFrameworks>
+        <TargetFrameworks Condition=" '$([MSBuild]::IsOsPlatform(`Windows`))' ">net472;net48;netcoreapp3.1;$(TargetFrameworks)</TargetFrameworks>
```
Comment (PrzemyslawKlys) [inline OfficeIMO.Tests/OfficeIMO.Tests.csproj:5]: I believe net core 3.1 also works on OSX/Linux so should be in all.

File: approve-all.ps1 [hunk 2]
```diff
@@ -0,0 +1,17 @@
+param (
```
Comment (PrzemyslawKlys) [inline approve-all.ps1:1]: I believe approve-all.ps1 it should go to Build folder. I know it's not there (not visible), but I do have some automation in place (just need to improve it for showing on GitHub), as it was just a temporary solution, that somehow stayed. 

image

**Comment by PrzemyslawKlys** [comment]
I've renamed the main pipeline. I guess it should work

image

File: OfficeIMO.Tests/OfficeIMO.Tests.csproj (hunk 1 above)
Comment (PrzemyslawKlys) [reply OfficeIMO.Tests/OfficeIMO.Tests.csproj:5]: I understand, and you're probably right. We can ignore 3.1, but then we need to remove it everywhere, not just in SLN.

**Comment by PrzemyslawKlys** [comment]
Is this ready? Or require some more work?

**Comment by rstm-sf** [comment]
Ok. It's done

Sorry for the delay, but I haven’t yet decided how to find the reason why the test breaks. But everything seems to be fine now (at least they passed several tests in a row). If the `AdvancedDocumentTests.AdvancedWordCreate` test flickers, you will have to skip it again and create a problem. Can tag me)

Also, I did not add all possible options to the tests. I think this is a good start

**Comment by PrzemyslawKlys** [comment]
Hi, 

Couple of questions:
- Do we need those csproj changes? 
- I saw you did remove a lot from gitignore? Not needed? Seems a lot got removed?
- Can't we reuse those tests we already have and simply add a line that is responsible for generation? 

I'm looking for number of line commits and it's 170k. If that has to happen with every change to library that seems like a lot of commits mostly because of .txt. 

I wonder if we're able to somehow solve it differently? 

Thank you,
Przemek

**Comment by PrzemyslawKlys** [comment]
I guess change to Azure Devops will break this table?

image

As it's no longer separate, so we won;t be able to tell?

**Comment by rstm-sf** [comment]
> Do we need those csproj changes?

Do you mean end of file? The current file the config, think the IDE fixed it itself https://github.com/EvotecIT/OfficeIMO/blob/2cf68efae04b4917ea911923a7e450463b8d4eab/.editorconfig#L5C1-L5C21

> I saw you did remove a lot from gitignore? Not needed? Seems a lot got removed?

Sorry, this is personal :) I can return it back. "GitIgnore for the studio" is more suitable for dense legacy...

> Can't we reuse those tests we already have and simply add a line that is responsible for generation?

It's possible, but I don't like how the tests are written. They assume that once launched, the application opens to view the result. This no longer looks like a unit test. In addition, there are no options (as far as I remember) that edit existing documents

> I'm looking for number of line commits and it's 170k. If that has to happen with every change to library that seems like a lot of commits mostly because of .txt.
>
> I wonder if we're able to somehow solve it differently?

It seems to me that this is true. Thus, we see all the changes that may affect with PR. The big change will be for new files. Well, now many tests have been added at once

For example, look at this commit https://github.com/EvotecIT/OfficeIMO/pull/86/commits/ca91747997b13b85f0958dcb195a43697013de2f

It shows that new styles were added last year, as well as endnotes and footnotes. Also, there are changes to previously generated properties. I think this is a very clear representation of how the file will change after PR

**Comment by PrzemyslawKlys** [comment]
> > Can't we reuse those tests we already have and simply add a line that is responsible for generation?
> 
> It's possible, but I don't like how the tests are written. They assume that once launched, the application opens to view the result. This no longer looks like a unit test. In addition, there are no options (as far as I remember) that edit existing documents
> 

Huh? I think we only have that option in Examples, not Tests? And it's always set to False? Not sure I understand?

**Comment by rstm-sf** [comment]
> Huh? I think we only have that option in Examples, not Tests? And it's always set to False? Not sure I understand?

Oops, my fail. 

But still, I don’t like that different approaches are mixed. If you want them to be in the same file/project, then I will delete them and add them to the existing ones

**Comment by PrzemyslawKlys** [comment]
It's fine, just trying to understand, that's all.

**Comment by PrzemyslawKlys** [comment]
I was just thinking it would mean for every new test, new file would be created as well, and that would lead to larger coverage, but maybe that's a bit too much and it's better to focus on directly those tests.

**Comment by rstm-sf** [comment]
> As it's no longer separate, so we won;t be able to tell?

Sorry, I didn't use azure devops as an administrator. Moreover, I did not generate badges for it. I think it’s worth returning back then (this table is not interesting for me :))

**Comment by rstm-sf** [comment]
Hello @PrzemyslawKlys

Are we going to merge this?)

Here we can see how the document changed during hold time (the tests did not change) https://github.com/EvotecIT/OfficeIMO/pull/86/commits/d4135d78e53c05098f83eef1374cad46642b6ff7

**Comment by PrzemyslawKlys** [comment]
Yes, we need to commit to this. I've been putting this away a bit, procrastinating because I know that once I commit this, I will have one more step to take care of with each and every PR ;)

**Pull request merged**